src/index.js

```diff
@@ -32,3 +32,7 @@ export const deepMap = _.curry((fn, obj, _map = map, is = isTraversable) =>
 // Misc
 // ----
 export const testRegex = regex => regex.test.bind(regex)
+
+// Gets all the regexp.exec indexes and their respective lastIndexes
+export const postings = _.curry((r, s) =>
+    _.map(x => (m => [m.index, r.lastIndex])(r.exec(s)), s.match(r)))
```

Comment:
Here's another one-liner. It passes al the tests and doesn't run `r.match`, but uses recursion:
```
export const postings = _.curry((r, s, a = []) =>
    (x => x ? postings(r, s, a.concat([[x.index, r.lastIndex]])) : a)(r.exec(s)))
```